File: python/utils/card_payment.py
# card_payment.py
class CardPayment:
    def __init__(self, card_number, expiration_date, cvv, amount):
        self.card_number = card_number
        self.expiration_date = expiration_date
        self.cvv = cvv
        self.amount = amount

    def pay(self):
        print(f"Card payment of {self.amount} made. Card Number: {self.card_number}, Expiration Date: {self.expiration_date}, CVV: {self.cvv}")

    def copy(self):
        return CardPayment(self.card_number, self.expiration_date, self.cvv, self.amount)

File: java/src/main/java/org/nitk/hackathon/utils/CardPayment.java
package org.nitk.hackathon.utils;

public class CardPayment implements Payment {
    private String cardNumber;
    private String expiryDate;
    private String cvv;

    public CardPayment(String cardNumber, String expiryDate, String cvv) {
        this.cardNumber = cardNumber;
        this.expiryDate = expiryDate;
        this.cvv = cvv;
    }

    @Override
    public void pay(double amount) {
        System.out.println("Paying " + amount + " using card " + cardNumber + " with expiry date " + expiryDate + " and cvv " + cvv);
    }

    @Override
    public Object copy() {
        return new CardPayment(cardNumber, expiryDate, cvv);
    }
}